binance-pipeline/lambdas/binance_historical_extraction.py

```diff
@@ -0,0 +1,249 @@
+# ./lambdas/binance_historical_extraction.py
+from datetime import datetime
+import json
+import math
+import time
+#
+from binance import BinanceQueryAPI
+from binance import TradeIDFinder, get_latest_trade_id, get_first_trade_id
+
+CHUNK_LIMIT = 1e3
+DATEFORMAT = "%Y-%m-%d"
+RATE_LIMIT: int = 12e3
+RESERVED_CONCURRENCY = 4
+SQS_QUEUE_NAME_EXTRACTION = ''
+SQS_QUEUE_NAME_DELAY_DAY = ''
+SQS_QUEUE_NAME_DELAY_HOUR = ''
+SQS_QUEUE_NAME_DELAY_MINUTE = ''
+SQS_QUEUE_NAME_DELAY_SECOND = ''
+
+def parse_date(date: str) -> datetime:
+
+    return datetime.strptime(date, DATEFORMAT)
+
+
+def parse_event(payload: dict):
+    if (type_ := payload['type']) == 'orchestration':
+        # user sent configuration
+        handle_orchestration(payload['config'])
+    elif type_ == 'api_call':
+        # sqs sent the payload to send as GET to API
+        handle_api_call(payload['config'])
+    else:
+        raise Exception("lambda only accepts api_call or orchestration")
+
+    return
+
+
+def ensure_sqs_trigger(record):
+    if (event_source := record["eventSource"]) != "aws:sqs":
+        raise Exception(f"Event source must be aws:sqs not {event_source = }")
```

Comment:
is the `=` in `{event_source = }` meant to be there?

Reply:
yes, as of Python 3.8 f strings can be printed with `f"{variable_name = }"` to produce `variable_name = %variable_value%`